ql/src/test/results/clientpositive/llap/constraints_optimization.q.out

```diff
@@ -2689,13 +2689,17 @@ POSTHOOK: Input: default@store_sales
 #### A masked pattern was here ####
 CBO PLAN:
 HiveProject(c_customer_id=[$1])
-  HiveAggregate(group=[{0, 1}])
-    HiveJoin(condition=[=($0, $8)], joinType=[inner], algorithm=[none], cost=[not available])
-      HiveProject(c_customer_sk=[$0], c_customer_id=[$1], c_first_name=[$8], c_last_name=[$9], c_preferred_cust_flag=[$10], c_birth_country=[$14], c_login=[$15], c_email_address=[$16])
-        HiveTableScan(table=[[default, customer]], table:alias=[customer])
-      HiveProject(ss_customer_sk=[$3])
-        HiveFilter(condition=[IS NOT NULL($3)])
-          HiveTableScan(table=[[default, store_sales]], table:alias=[store_sales])
+  HiveSemiJoin(condition=[=($0, $2)], joinType=[semi])
```

Comment:
We should not trigger the optimization in this case, since it is just doing consecutively the same join. However, this is not actually related to this patch. Can we create a follow-up?